pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -1359,13 +1372,44 @@ func (r *ReconcilePerconaServerMongoDB) reconcileStatefulSet(
 		return nil, errors.Wrapf(err, "PodDisruptionBudget for %s", sfs.Name)
 	}
 
+	if err := r.reconcilePVCs(sfs, matchLabels); err != nil {
+		return nil, errors.Wrapf(err, "reconcile PVCs for %s", sfs.Name)
+	}
+
 	if err := r.smartUpdate(cr, sfs, replset); err != nil {
 		return nil, errors.Wrap(err, "failed to run smartUpdate")
 	}
 
 	return sfs, nil
 }
 
+func (r *ReconcilePerconaServerMongoDB) reconcilePVCs(sfs *appsv1.StatefulSet, ls map[string]string) error {
+	pvcList := &corev1.PersistentVolumeClaimList{}
+	err := r.client.List(context.TODO(), pvcList, &client.ListOptions{
+		Namespace:     sfs.Namespace,
+		LabelSelector: labels.SelectorFromSet(ls),
+	})
+	if err != nil {
+		return errors.Wrap(err, "list PVCs")
+	}
+
+	for _, pvc := range pvcList.Items {
+		if compareMaps(sfs.Labels, pvc.Labels) {
+			continue
+		}
+
+		orig := pvc.DeepCopy()
+		pvc.Labels = sfs.Labels
```

Comment:
What about labels that was already in `pvc.Labels`? Maybe we should leave them?

Reply:
yes, I think we should: https://github.com/percona/percona-server-mongodb-operator/pull/881/commits/2fcb20529d835adc2872354a3e64bd3974d59dd2